### PR TITLE
fix: make `#file` variables part of the chat editing working set

### PR DESCRIPTION
**Pull request by joyceerhl**
cc @roblourens @jrieken 

I want to avoid rerendering the editing widget too frequently (i.e. not re-parsing and rerendering on input editor content change), and I also don't want to start refcounting variable references in the input part, so I went with forcing a re-parse of the input request whenever we accept/remove a dynamic completion and when the input is cleared, which I think covers the majority of cases. LMK if you have questions or suggestions

### CODE REVIEW

**Review by DonJayamanne — APPROVED** [review]

File: src/vs/workbench/contrib/chat/browser/chatWidget.ts [hunk 1]
```diff
@@ -460,6 +471,11 @@ export class ChatWidget extends Disposable implements IChatWidget {
 		return this.inputPart.hasFocus();
 	}
 
+	refreshParsedInput() {
+		this.parsedChatRequest = this.instantiationService.createInstance(ChatRequestParser).parseChatRequest(this.viewModel!.sessionId, this.getInput(), this.location, { selectedAgent: this._lastSelectedAgent });
+		this._onDidChangeParsedInput.fire();
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/browser/chatWidget.ts:476]: Where is this being listened to in your scenario? I think I see the problem but I don't really like this pattern of what might look like triggering this from a random subset of the places that lead to changes in the result.

Comment (joyceerhl) [reply src/vs/workbench/contrib/chat/browser/chatWidget.ts:476]: https://github.com/microsoft/vscode/blob/63e5a64a61d75b5f2f26806939ef9c853881d729/src/vs/workbench/contrib/chat/browser/chatWidget.ts#L277-L279

Comment (roblourens) [reply src/vs/workbench/contrib/chat/browser/chatWidget.ts:476]: And you basically just want to know when a `#file` showed up in the query? Would it suffice to have some other event to listen to when that changes?

Comment (joyceerhl) [reply src/vs/workbench/contrib/chat/browser/chatWidget.ts:476]: Or when a `#file` disappears from the query

Comment (roblourens) [reply src/vs/workbench/contrib/chat/browser/chatWidget.ts:476]: Right, maybe you can have an event on `ChatDynamicVariableModel` to listen to, or maybe these should live in the attachment model somehow. Adding an item to my debt list.

**Pull request merged**